### PR TITLE
Use a prototype setter instead of defineProperty for currentTag

### PR DESCRIPTION
**Pull request by garlicnation**

### CODE REVIEW

File: externs/customelements.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/** @constructor */
+var CustomElementsRegistry;
+//
+// /** @type string */
```
Comment (justinfagnani) [inline externs/customelements.js]: uncomment lines

File: externs/customelements.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+CustomElementsRegistry.prototype.currentTag;
```
Comment (justinfagnani) [inline externs/customelements.js]: Add the other public properties that I exported via property definitions.

File: src/CustomElements/v1/CustomElements.js [hunk 3]
```diff
@@ -61,27 +61,18 @@ var CustomElementDefinition;
...
-    this._attributeObserver =
-        new MutationObserver(this._handleAttributeChange.bind(this));
-    this._newInstance = null;
-    this._newTagName = null;
-    this.polyfilled = true;
-  }
-
-  /** @lends {CustomElementsRegistry.prototype} */
-  CustomElementsRegistry.prototype = {
+  /** @export */
```
Comment (justinfagnani) [inline src/CustomElements/v1/CustomElements.js]: Remove. `@export` doesn't do anything, without `--generate_exports`, and then it requires runtime support.

File: src/CustomElements/v1/CustomElements.js [hunk 4]
```diff
@@ -271,21 +264,17 @@ var CustomElementDefinition;
...
-  // TODO: Figure out how to export a setter w/o using defineProperty
-  Object.defineProperty(CustomElementsRegistry.prototype, 'currentTag', {
-    set(tagName) {
-      this._newTagName = this._newTagName || tagName;
-    },
-  });
-
   // Closure Compiler Exports
   window['CustomElementsRegistry'] = CustomElementsRegistry;
   CustomElementsRegistry.prototype['define'] = CustomElementsRegistry.prototype.define;
```
Comment (justinfagnani) [inline src/CustomElements/v1/CustomElements.js]: remove this and the following two lines

**Comment by justinfagnani** [comment]
I think this is worth merging, even though it compiles to defineProperties. It only adds about 40 bytes to the gzipped output.

**Comment by TimvdLippe** [comment]
Closing this PR due to age.

**Pull request closed without merging**